rest/handler.go

```diff
@@ -254,3 +281,45 @@ func getLimitFromURL(url *url.URL) (bool, int64, error) {
 
 	return true, limit, nil
 }
+
+type urlHandler struct {
+	caller func(w http.ResponseWriter, r *http.Request, s langserver.HeadlessServer, requestID string)
+	h      *langserverHandler
```

Comment:
This type is unnecessary since every time we work with a URL handler (except `ServeHTTP`) the respective `langserverHandler` is known.

Reply:
The whole code could be made simpler now by just instrumenting languageServer from the original code. This type was created because i assume the metrics endpoint doesn’t need to be instrumented so hence i created separate handlers. sorry for the mix up!

Reply:
one issue though, this type help add labels to handlers. for instance without a separate handler(urlHandler) for each endpoint we would end up instrumenting the handler to give metrics like so http_request_duration_seconds_bucket{code="200",handler="languageServer",le="0.005"}  for all requests instead of having metrics like http_request_duration_seconds_bucket{code="200",handler="hover",le="0.005"} 
http_request_duration_seconds_bucket{code="200",handler="completion",le="0.005"} 
to handle various request

Reply:
I don't understand.

What would be the problem with passing what is now called `caller` to `langserverHandler.Handle`?

Reply:
sorry for the late reply . 
we are using [InstrumentHandlerDuration](https://godoc.org/github.com/prometheus/client_golang/prometheus/promhttp#InstrumentHandlerDuration) which requires a handler type as one of its argument.
the caller function is not of type handler so we can't pass it to langserverHandler.Handle.

Reply:
That makes sense.

Then just add a comment explaining this and maybe give it a more descriptive name like `subHandler`.

Reply:
alright!